### PR TITLE
Re-title a related-articles section in “Iteration protocols” doc

### LINKED ISSUE #17540: Reference | Iteration Protocols - has multiple sections of the same name
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols

### What specific section or headline is this issue about?

See Also

### What information was incorrect, unhelpful, or incomplete?

There are two See Also sections.

### What did you expect to see?

These sections likely should have been merged.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/reference/iteration_protocols`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/iteration_protocols/index.md
* Last commit: https://github.com/mdn/content/commit/4d7aa86ff44bfe23d1a4b60a2c3aafb25f9a118a
* Document last modified: 2022-06-20T13:20:39.000Z

</details>

### PR DESCRIPTION
**Pull request by sideshowbarker**
Fixes https://github.com/mdn/content/issues/17540. Also fixes some unrelated typos.

### CODE REVIEW

File: files/en-us/web/javascript/reference/iteration_protocols/index.md [hunk 1]
```diff
@@ -195,14 +195,6 @@ new WeakSet(
...
 ```
 
-#### See also
-
-- {{jsxref("Promise.all()", "Promise.all(<var>iterable</var>)")}}
-- {{jsxref("Promise.allSettled()", "Promise.allSettled(<var>iterable</var>)")}}
-- {{jsxref("Promise.race()", "Promise.race(<var>iterable</var>)")}}
-- {{jsxref("Promise.any()", "Promise.any(<var>iterable</var>)")}}
-- {{jsxref("Array.from()", "Array.from(<var>iterable</var>)")}}
-
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/iteration_protocols/index.md:205]: Moving this to the bottom makes it out of context and much less relevant. What about merging it with the list above instead? I'm not sure why these in a separate section.

Comment (sideshowbarker) [reply files/en-us/web/javascript/reference/iteration_protocols/index.md]: > Moving this to the bottom makes it out of context and much less relevant.

I guess that’s true, yeah

> What about merging it with the list above instead?

Given that the docs follow a structural convention of having a See Also section at the end, moving an existing See Also out of the end of that doc would be a break with the normal structural conventions.

So I (force) pushed an update that re-titles the first list to “Other articles relevant to built-in iterables”. Lemme know what you think.

Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/iteration_protocols/index.md]: I was pointing to the "There are many APIs that accept iterables. Some examples include:" section above. We can either append these to that list, or we can change this heading to a paragraph like "Other APIs accepting iterables include:", although I'm not sure why we need two separate lists in the first place.

Comment (sideshowbarker) [reply files/en-us/web/javascript/reference/iteration_protocols/index.md]: > I'm not sure why we need two separate lists in the first place.

We don’t need two separate lists. So combining them is fine

> I was pointing to the "There are many APIs that accept iterables. Some examples include:" section above. We can either append these to that list, or we can change this heading to a paragraph like "Other APIs accepting iterables include:"

Either of those sound fine to me. Can you maybe open a separate new PR that makes whichever change seems better to you? And then we could close this PR as superseded by that one.

File: files/en-us/web/javascript/reference/iteration_protocols/index.md [hunk 2]
```diff
@@ -195,7 +195,7 @@ new WeakSet(
 ).has(myObj); // true
 ```
 
-#### See also
+#### Other articles relevant to built-in iterables
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/iteration_protocols/index.md:198]:
```suggestion
Other built-in APIs that accept iterables include:
```

My idea is a simple change like that😄 No need for another PR

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**